### ISSUE TITLE
HTTP/3

### ISSUE THREAD

**Issue by splitice** [openresty/openresty#741]
Any statement @agentzh on whether HTTP/3 will be on the Roadmap when nginx-quic is production ready?

**Comment by inevity** [comment]
please see https://www.nginx.com/blog/our-roadmap-quic-http-3-support-nginx/

**Comment by LoganDark** [comment]
OpenResty seems to be a pretty tricked out Nginx distribution, so I'm surprised that HTTP3 is still not supported. Since it's Q2 2022, what's the blocker right now? The fact that it's not merged into mainline yet or..?

**Comment by DevHaoZi** [comment]
any update?

**Comment by splitice** [comment]
HTTP/3 just merged with 1.25.0

**Comment by splitice** [comment]
To everyone celebrating, just a reminder that openresty only offers very basic support for http/2 years after it merged with nginx. Celebrate when this issue even gets a response... Not before.

**Comment by laiqinghua** [comment]
那岂不是马上就有HTTP3可以使用了？

**Comment by clywm520** [comment]
nginx-1.25.0 已经支持http3.0.  openrestry什么时候支持呢

**Comment by zhuizhuhaomeng** [comment]
这个需要大家来贡献力量，共同推进

**Comment by splitice** [comment]
Related comment https://github.com/openresty/lua-nginx-module/pull/2210#issuecomment-1668058231

@swananan has made the first steps in this direction 🎉

**Comment by juberstine-godaddy** [comment]
Any update on this?

**Comment by zhuizhuhaomeng** [comment]
Just wait for good news.

**Comment by manoaratefy** [comment]
The latest release of OpenResty is now based on 1.25.3. However, there's a little fix to really implement HTTP/3: the .rpm provided package has been compiled without `--with-http_v3_module`. I guess all packages are same. I think this should be added before we can claim a HTTP/3 support.

**Comment by selboo** [comment]
And pcre2 is not enabled by default

https://github.com/openresty/openresty/blob/master/util/configure#L558

```
    # disable pcre2 by default
    push @ngx_opts, '--without-pcre2';
```

**Comment by devhaozi** [comment]
> And pcre2 is not enabled by default
> 
> https://github.com/openresty/openresty/blob/master/util/configure#L558
> 
> ```
>     # disable pcre2 by default
>     push @ngx_opts, '--without-pcre2';
> ```

fixed in #955 #956

**Comment by koehn** [comment]
There will also need to be an OpenResty project for the `http_v3` module so that it can be linked in. At least, that’s how [the other modules](https://github.com/openresty?q=module&type=all&language=&sort=) are handled.

**Comment by Lathanderjk** [comment]
Any news?, HTTP 3 is on 1/3 website and openresty has no support for it. Also nginx 1.27.0 is out for few weeks now, HTTP/3 support should be mature enough at least for initial support.

**Comment by ivanjx** [comment]
@Lathanderjk the docker openresty already supports nginx if you havent noticed

**Comment by devhaozi** [comment]
OpenResty 1.27.1.1 has been released and this issue can be closed.
